
syntax = "proto2";
package aos.record.proto;
// Next ID: 3
message ProtoDesc {
  optional bytes cc_proto = 1;
  optional bytes py_proto = 2;
}
// Next ID: 4
message TopicMsgDesc {
  optional string topic_name = 1;
  // message_type full_name. like: aos.record.proto.TopicMsgDesc
  optional string message_type = 2;
  // proto file destriptions, it used to register Message
  // and Deserialzed message proto bytes.
  optional ProtoDesc proto_desc = 3;
};

message TransportParams {
  optional string topic_name = 1;
}

// Next ID: 4
message Header {
  optional int32 id = 1;
  optional double timestamp = 2;
  optional TransportParams transport_params = 3;
}